gui-libs/display-manager-init/files/display-manager.initd-11

```diff
@@ -0,0 +1,235 @@
+# /usr/bin/startDM after the current runlevel completes (this
+# script should only be added to the actual runlevel the user is
+# using).
+#
+# Martin Schlemmer
+# aka Azarah
+# 04 March 2002
+
+depend() {
+	need localmount display-manager-setup
+
+	# this should start as early as possible
+	# we can't do 'before *' as that breaks it
+	# (#139824) Start after ypbind and autofs for network authentication
+	# (#145219 #180163) Could use lirc mouse as input device
+	# (#70689 comment #92) Start after consolefont to avoid display corruption
+	# (#291269) Start after quota, since some dm need readable home
+	# (#390609) gdm-3 will fail when dbus is not running
+	# (#366753) starting keymaps after X causes problems
+	after bootmisc consolefont modules netmount
+	after readahead-list ypbind autofs openvpn gpm lircmd
+	after quota keymaps
+	before alsasound
+
+	# Start before X
+	use consolekit dbus xfs
+}
+
+setup_dm() {
+	local MY_XDM
+	
+	MY_XDM=$(echo "${DISPLAYMANAGER}" | tr '[:upper:]' '[:lower:]')
+
+	# Load our root path from profile.env
+	# Needed for kdm
+	PATH=${PATH}:$(. /etc/profile.env; echo "${ROOTPATH}")
+
+	NAME=
+	case "${MY_XDM}" in
+		kdm|kde)
+			EXE=/usr/bin/kdm
+			PIDFILE=/run/kdm.pid
+			;;
+		entrance*)
+			EXE=/usr/sbin/entrance
+			PIDFILE=/run/entrance.pid
+			;;
+		gdm|gnome)
+			# gdm-3 and above has different paths
+			if [ -f /usr/sbin/gdm ]; then
+				EXE=/usr/sbin/gdm
+				PIDFILE=/run/gdm/gdm.pid
+				START_STOP_ARGS="--background"
+				AUTOCLEAN_CGROUP="yes"
+			else
+				EXE=/usr/bin/gdm
+				PIDFILE=/run/gdm.pid
+			fi
+			[ "${RC_UNAME}" != "Linux" ] && NAME=gdm-binary
+			;;
+		greetd)
+			EXE=/usr/bin/greetd
+			START_STOP_ARGS="-m --background"
+			PIDFILE=/run/greetd.pid
+			;;
+		wdm)
+			EXE=/usr/bin/wdm
+			PIDFILE=
+			;;
+		gpe)
+			EXE=/usr/bin/gpe-dm
+			PIDFILE=/run/gpe-dm.pid
+			;;
+		lxdm)
+			EXE=/usr/sbin/lxdm-binary
+			PIDFILE=/run/lxdm.pid
+			START_STOP_ARGS="--background"
+			;;
+		lightdm)
+			EXE=/usr/sbin/lightdm
+			PIDFILE=/run/lightdm.pid
+			START_STOP_ARGS="--background"
+			;;
+		sddm)
+			EXE="/usr/bin/sddm"
+			START_STOP_ARGS="-m --background"
+			PIDFILE=/run/sddm.pid
+			;;
+		*)
+			# first find out if there is such executable
+			EXE="$(command -v ${MY_XDM} 2>/dev/null)"
+			PIDFILE="/run/${MY_XDM}.pid"
+
+			# warn user that they are doing sick things if the exe was not found
+			if [ -z "${EXE}" ]; then
+				echo "ERROR: Your XDM value is invalid."
+				echo "  No ${MY_XDM} executable could be found on your system."
+			fi
+			;;
+	esac
+
+	if ! [ -x "${EXE}" ]; then
+		EXE=/usr/bin/xdm
+		PIDFILE=/run/xdm.pid
+		if ! [ -x "/usr/bin/xdm" ]; then
+			echo "ERROR: Please set your DISPLAYMANAGER variable in /etc/conf.d/xdm,"
+			echo "	or install x11-apps/xdm package"
+			eend 255
+		fi
+	fi
+}
+
+# Check to see if something is defined on our VT
+vtstatic() {
+	if [ -e /etc/inittab ] ; then
+		grep -Eq "^[^#]+.*\<tty$1\>" /etc/inittab
```

Comment:
This is not the right way to do it. This only validate if inittab spawned anything on TTY. I myself often use openvt to spawn some interactive processes on VT7 and above. If you want bool context to check if tty is free, this will work for you:

```
openvt -c 7 true >/dev/null 2>&1
```

It will give you exit status 0 for not in use TTY, and non-zero for already in-use TTY.